src/Test/Utilities/Portable/Metadata/ILValidation.cs

```diff
@@ -3,16 +3,182 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.IO;
 using System.Linq;
+using System.Reflection;
 using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Security.Cryptography;
 using System.Text;
+using System.Threading;
 using System.Xml;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis;
 using Microsoft.Metadata.Tools;
 
 namespace Roslyn.Test.Utilities
 {
     public static class ILValidation
     {
+        /// <summary>
+        /// Validates that the given stream is marked as signed, the signature matches
+        /// the public key, and the header checksum is correct.
+        /// </summary>
+        public static bool IsStreamFullSigned(Stream moduleContents)
+        {
+            moduleContents.Position = 0;
```

Comment:
This is a visible mutation to the stream. Think this method should restore the original position upon exit.